target/linux/ramips/image/mt7620.mk

```diff
@@ -7,16 +7,13 @@ define Build/tplink-header
 		-o $@.new -k $@ -r $(IMAGE_ROOTFS) && mv $@.new $@
 endef
 
-define Build/pad-kernel-ex2700
-	cp $@ $@.tmp && dd if=/dev/zero bs=64 count=1 >> $@.tmp \
-		&& dd if=$@.tmp of=$@.new bs=64k conv=sync && truncate -s -64 $@.new \
-		&& cat ex2700-fakeroot.uImage >> $@.new && rm $@.tmp && mv $@.new $@
-endef
-
-define Build/netgear-header
-	$(STAGING_DIR_HOST)/bin/mkdniimg \
-		$(1) -v OpenWrt -i $@ \
-		-o $@.new && mv $@.new $@
+define Build/netgear-kernel
+	let \
+		bsize="$(subst k,*1024,$(1))" \
+		size="$$(stat -c%s '$@')" \
+		size="((1 + ((64 + size) / bsize)) * bsize) - 64"; \
+	dd if=$@ of=$@.new bs=$$size count=1 conv=sync \
```

Comment:
Thanks a lot for the explanation. If I got it right, you need something like ```pad-to 65472```. But since it is kinda hard to read I would suggest to use ```pad-offset 64k 64``` which should create exactly the padding you need.

Now that I know that netgear-fakeroot.uImage  is just a fake uImage header, I would suggest to rename this file to something like netgear-fake-uImage-header. But that is really something optional and you might find even a better name.